src/applications/gi/components/profile/SchoolLocations.jsx

```diff
@@ -0,0 +1,196 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import { getCalculatedBenefits } from '../../selectors/calculator';
+
+const DEFAULT_ROWS_VIEWABLE = 10;
+const DEFAULT_ROWS_ADJUSTED = DEFAULT_ROWS_VIEWABLE - 1;
+
+export class SchoolLocations extends React.Component {
+  static propTypes = {
+    institution: PropTypes.object,
+    facilityMap: PropTypes.object,
+    calculator: PropTypes.object,
+    constants: PropTypes.object,
+  };
+
+  constructor(props) {
+    super(props);
+    this.state = { viewMore: false };
+  }
+
+  institutionIsBeingViewed = facilityCode =>
+    facilityCode === this.props.institution.facilityCode;
+
+  shouldHideViewMore = (branches, extensions) => {
+    let totalRows = 1 + branches.length + extensions.length; // always has a main row
+    branches.forEach(branch => {
+      totalRows += branch.extensions.length;
+    });
+    return totalRows > DEFAULT_ROWS_VIEWABLE && !this.state.viewMore;
+  };
+
+  createLinkTo = (facilityCode, name) => {
+    if (this.institutionIsBeingViewed(facilityCode)) {
+      return name;
+    }
+    const { version } = this.props;
+    const query = version ? `?version=${version}` : '';
+
+    return (
+      <a href={`${facilityCode}${query}`}>
+        <h6>{name}</h6>
+      </a>
+    );
+  };
+
+  handleViewMoreClicked = () => {
+    this.setState({ ...this.state, viewMore: true });
```

Comment:
`setState` merges the new values with the existing state so `...this.state` shouldn't be necessary.

Reply:
changed